packages/cli/src/commander.ts

```diff
@@ -0,0 +1,255 @@
+}
+
+function linkCmd() {
+  return createCommand('link')
+    .description('Link the given Mermaid diagram to Mermaid Chart')
+    .addArgument(new Argument('<path>', 'The path of the file to link.'))
+    .action(async (path, _options, command) => {
+      const optsWithGlobals = command.optsWithGlobals<CommonOptions>();
+      const client = await createClient(optsWithGlobals);
+      const linkCache = {};
+
+      const existingFile = await readFile(path, { encoding: 'utf8' });
+
+      const linkedDiagram = await link(existingFile, client, {
+        cache: linkCache,
+        title: path,
+        async getProjectId(cache) {
+          cache.projects = cache.projects ?? client.getProjects();
+          const projectId = await select({
+            message: `Select a project to upload ${path} to`,
+            choices: [
+              ...(await cache.projects).map((project) => {
+                return {
+                  name: project.title,
+                  value: project.id,
+                };
+              }),
+              new Separator(
+                `Or go to ${new URL('/app/projects', client.baseURL)} to create a new project`,
+              ),
+            ],
+          });
+          return projectId;
+        },
+      });
+
+      await writeFile(path, linkedDiagram, { encoding: 'utf8' });
+    });
+}
+
+function pullCmd() {
+  return createCommand('pull')
+    .description('Pulls a document from from Mermaid Chart')
+    .addArgument(new Argument('<path>', 'The path of the file to pull.'))
+    .option('--check', 'Check whether the local file would be overwrited')
+    .action(async (path, options, command) => {
+      const optsWithGlobals = command.optsWithGlobals<CommonOptions>();
+      const client = await createClient(optsWithGlobals);
+      const text = await readFile(path, { encoding: 'utf8' });
+
+      const newFile = await pull(text, client, { title: path });
+
+      if (text === newFile) {
+        console.log(`✅ - ${path} is up to date`);
+      } else {
+        if (options['check']) {
+          console.log(`❌ - ${path} would be updated`);
+          process.exitCode = 1;
+        } else {
+          await writeFile(path, newFile, { encoding: 'utf8' });
+          console.log(`✅ - ${path} was updated`);
+        }
+      }
+    });
+}
+
+function pushCmd() {
+  return createCommand('push')
+    .description('Push a local diagram to Mermaid Chart')
+    .addArgument(new Argument('<path>', 'The path of the file to push.'))
+    .action(async (path, _options, command) => {
+      const optsWithGlobals = command.optsWithGlobals<CommonOptions>();
+      const client = await createClient(optsWithGlobals);
+      const text = await readFile(path, { encoding: 'utf8' });
+
+      await push(text, client, { title: path });
+    });
+}
+
+export function createCommanderCommand() {
+  const program = createCommand('mermaid-cli')
+    .version('0.1.0-alpha.0') // TODO: how can we keep this synced with package.json
```

Comment:
Can't we import it from the json file? That's how we do in mermaid.

Reply:
Fixed by https://github.com/Mermaid-Chart/plugins/pull/12/commits/480df29378136f2d3636c78fe03d9c00fac1f717

I think in a previous version of this library, I had the file in `src/index.ts`, but I compiled the output into `./index.js`. So `require("../package.json");` would have had to have switched to `require("./package.json");`. But now I'm using a `dist/` folder, so the import path to `package.json` is the same!